subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -2849,13 +2850,41 @@ static int nrf_cloud_parse_location_json(const cJSON *const loc_obj,
 			location_out->type = LOCATION_TYPE_SINGLE_CELL;
 		} else if (!strcmp(type, NRF_CLOUD_LOCATION_TYPE_VAL_WIFI)) {
 			location_out->type = LOCATION_TYPE_WIFI;
+		} else if (!strcmp(type, NRF_CLOUD_LOCATION_TYPE_VAL_ANCHOR)) {
+			location_out->type = LOCATION_TYPE_WIFI;
+			anchor = true;
 		} else {
 			LOG_WRN("Unhandled location type: %s", type);
 		}
 	} else {
 		LOG_WRN("Location type not found in message");
 	}
 
+	/* Print anchor info */
+	if (anchor) {
+		cJSON *anchors, *mac, *name;
+		int anc_cnt;
+
+		/* Anchor info is provided in an array of objects */
+		anchors	= cJSON_GetObjectItem(loc_obj, NRF_CLOUD_LOCATION_JSON_KEY_ANCHORS);
+		anc_cnt = cJSON_GetArraySize(anchors);
+
+		for (int idx = 0; idx < anc_cnt; ++idx) {
+			cJSON *anc = cJSON_GetArrayItem(anchors, idx);
+
+			mac	= cJSON_GetObjectItem(anc, NRF_CLOUD_LOCATION_JSON_KEY_ANC_MAC);
+			name	= cJSON_GetObjectItem(anc, NRF_CLOUD_LOCATION_JSON_KEY_ANC_NAME);
```

Comment:
Maybe not for this PR, but it seems like perhaps `location_out` should be expanded to accept an `anchor_name` field of some kind. Perhaps optionally, or maybe if we restrict the anchor name size, it could be a buffer directly stored in the struct. IDK.

But I do absolutely see potential use cases for named locations in firmware.

Reply:
Yes.